packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -675,6 +675,15 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         let channelBaseGCDetails = (await this.channelsBaseGCDetails).get(channelId);
         if (channelBaseGCDetails === undefined) {
             channelBaseGCDetails = {};
+        } else if (channelBaseGCDetails.gcData?.gcNodes !== undefined) {
```

Comment:
I just realized that this bug was a regression added by this PR - https://github.com/microsoft/FluidFramework/pull/8389. Take a look at `this.initialChannelGCDataP` which was removed / updated by the above PR from `dataStoreRuntime.ts`. It called `removeRouteFromAllNodes(gcData.gcNodes, this.absolutePath);` to remove the back routes which was removed :-(

Reply:
nit: You could also do what was happening before this bug was introduced, i.e., call `removeRouteFromAllNodes(gcData.gcNodes, this.absolutePath)` in `this.channelsBaseGCDetails` after getting the baseGCDetails which is initialized in the data store runtime constructor. That should be a one line change.